keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -491,6 +512,34 @@ TypedValue](https://github.com/kubernetes/kubernetes/blob/9ff3b7e744b34c099c1405
 object has any unknown or duplicate fields.
 
 
+#### Kubectl Flag
+
+For beta, we will modify the kubectl `--validate` flag from being a bool flag to
+a string flag that accepts the following values:
+
+* `server`: This sends a request with the fieldValidation param set to `Strict`
+for strict server-side validation.
+* `client`: This performs client-side validation in kubectl that breaks if it
+encounters any validation errors before sending a request to the server. If no
+client-side validation errors are found, the request sent to the server has
+fieldValidation param set to `Ignore`. This will be the default behavior in beta
+as it is identical to the current validation behavior pre-1.24.
```

Comment:
given the problems with client-side validation, I actually don't want to **increase** the ability to specifically request it... keeping the flag intent-based (`--validate=true|false|strict|warn|ignore`) and using server-side validation when available makes more sense to me

Reply:
see comment below. I'm hesitant to remove the ability to explicitly request client-side validation on ss-validation enabled servers before discussing with sig-cli